File: GFG/Heap/maxitems.py
import ds

def maxit(arr,k):
    mih=ds.MinHeap()
    for i in arr:
        mih.insert(i)
    i=0
    while(k>0):
        
        temp=mih.extract_min()
        if(k<temp):
            break
        print(temp,end=" ")
        i+=1
        k=k-temp
    print()
    print(i)
    
arr=[1,12,5,111,200]
suma=100
maxit(arr,suma)

File: GFG/Heap/maxitems.cpp
#include<iostream>
#include"../ds.h"
using namespace std;

void maxit(vector<int> &arr,int suma){
    MinHeap<int> mih;
    for(auto i:arr){
        mih.insert(i);
    }
    int i=0;
    while(suma>=0){
        int temp=mih.extractMin();
        if(suma<temp){
            break;
        }
        suma=suma-temp;
        i++;
        cout<<temp<<" ";
    }
    cout<<"\n"<<i;
    return;
}

int main(){
    vector<int> arr {1,12,5,111,200};
    int k=250;
    maxit(arr,k);
    return 0;
}